### PR TITLE
Update class lookup and instantiating with varargs style

### PR DESCRIPTION
**Pull request by makiam**
More array to varargs style call changes

### CODE REVIEW

**Comment by peastman** [comment]
Nice, definitely cleaner that way.

File: ArtOfIllusion/src/artofillusion/procedural/Module.java [hunk 1]
```diff
@@ -285,11 +286,15 @@ public boolean edit(BFrame fr, Scene theScene)
...
 
-  /** This method initializes the module in preparation for evaluating the procedure at a
-      new point.  The default implementation does nothing.  Subclasses whose output depends
-      on the point should override this method. */
+  /**
+   * This method initializes the module in preparation for evaluating the
+   * procedure at a new point.The default implementation does nothing.
+   * Subclasses whose output depends on the point should override this method.
+   *
+   * @param point
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/procedural/Module.java]: @param annotations in the javadoc should add information beyond what can be discerned from the type signature. Non-obvious use details, etc.

File: Tests/src/artofillusion/procedural/ModuleTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (Sailsman63) [inline Tests/src/artofillusion/procedural/ModuleTest.java]: Please do add a proper license header when creating a new file. It helps with clarity, and I've heard that in some jurisdictions, lack of a license is interpreted in such a way that @peastman or I could (theoretically) face legal difficulties if we added the file to the project.

**Comment by Sailsman63** [comment]
I do like this style for these sorts of cases!

File: ArtOfIllusion/src/artofillusion/procedural/Module.java [hunk 3]
```diff
@@ -332,17 +338,18 @@ public void getColor(int which, RGBColor color, double blur)
...
   {
     try
     {
-      Constructor con = getClass().getConstructor(new Class [] {Point.class});
-      return (Module) con.newInstance(new Object [] {new Point(bounds.x, bounds.y)});
+      Constructor con = getClass().getConstructor(Point.class);
+      return (Module) con.newInstance(new Point(bounds.x, bounds.y));
     }
-    catch (Exception ex)
+    catch (IllegalAccessException | IllegalArgumentException | InstantiationException | NoSuchMethodException | SecurityException | InvocationTargetException ex)
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/procedural/Module.java]: I've already merged, and then had to revert this in master.

How did you get the multi-catch to compile? It's Java7+ and the ant file locks in at Java6

Comment (peastman) [reply ArtOfIllusion/src/artofillusion/procedural/Module.java]: Aren't we now using Java 8 as our minimum?

Comment (Sailsman63) [reply ArtOfIllusion/src/artofillusion/procedural/Module.java]: I thought that we had decided to wait that until releasing the latest bugfixes (Possibly my misunderstanding)

The build files still reference `source 1.6`, so this should not have compiled...

**Pull request merged**